apps/njtransit/nj_transit.star

```diff
@@ -0,0 +1,374 @@
+                color = "#666",
+            ),
+            rendered[1],
+        ],
+    )
+
+"""
+Creates a Row and adds needed children objects
+for a single departure.
+"""
+
+def render_departure_row(departure):
+    background_color = render.Box(width = 22, height = 11, color = COLOR_MAP.get(departure.service_line, DEFAULT_COLOR))
+    destination_text = render.Marquee(
+        width = 36,
+        child = render.Text(departure.destination, font = "Dina_r400-6", offset = -2, height = 7),
+    )
+
+    departing_in_text = render.Text(departure.departing_in, color = "#f3ab3f")
+
+    #If we have a Track Number append and make it a scroll marquee
+    if departure.track_number != None:
+        depart = "{} - Track {}".format(departure.departing_in, departure.track_number)
+        departing_in_text = render.Marquee(
+            width = 36,
+            child = render.Text(depart, color = "#f3ab3f"),
+        )
+
+    if departure.departing_in.startswith("at"):
+        departing_in_text = render.Marquee(
+            width = 36,
+            child = render.Text(departure.departing_in, color = "#f3ab3f"),
+        )
+
+    child_train_number = render.Text(departure.train_number, font = "CG-pixel-4x5-mono")
+
+    if len(departure.train_number) > 4:
+        child_train_number = render.Marquee(child = child_train_number)
+
+    train_number = render.Box(
+        color = "#0000",
+        width = 22,
+        height = 11,
+        child = child_train_number,
+    )
+
+    stack = render.Stack(children = [
+        background_color,
+        train_number,
+    ])
+
+    column = render.Column(
+        children = [
+            destination_text,
+            departing_in_text,
+        ],
+    )
+
+    return render.Row(
+        expanded = True,
+        main_align = "space_evenly",
+        cross_align = "center",
+        children = [
+            stack,
+            column,
+        ],
+    )
+
+def get_schema():
+    options = getStationListOptions()
+
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Dropdown(
+                id = "station",
+                name = "Departing Station",
+                desc = "The NJ Transit Station to get departure schedule for.",
+                icon = "train",
+                default = options[0].value,
+                options = options,
+            ),
+        ],
+    )
+
+"""
+Function gets all depatures for a given station
+returns a list of structs with the following fields
+
+depature_item struct:
+    departing_at: string
+    destination: string
+    service_line: string
+    train_number: string
+    track_number: string
+    departing_in: string
+"""
+
+def get_departures_for_station(station):
+    print("Getting departures for '%s'" % station)
+
+    station_suffix = station.replace(" ", "%20")
+    station_url = "{}/{}".format(NJ_TRANSIT_DV_URL, station_suffix)
+
+    print(station_url)
+
+    nj_dv_page_response = http.get(station_url)
```

Comment:
Would you mind adding some caching here? Otherwise we would be hammering the NJ transit website if this app gets popular :scream_cat:

Reply:
Done added 1min ttl for website results.